### PR TITLE
Add automatic server icon change every 24-48h

### PR DESCRIPTION
**Pull request by ivandardi**

### CODE REVIEW

**Review by anden3 — APPROVED** [review]
Looks good, although I'd probably store the current path in a variable outside the loop so it can be filtered out to avoid icon reuse, as long as there's more than one icon that is.

File: src/main.rs [hunk 1]
```diff
@@ -209,5 +208,57 @@ async fn event_handler(
...
+		let http = ctx.http.clone();
+		tokio::spawn(init_server_icon_changer(http, data.discord_guild_id));
+	}
+
 	Ok(())
 }
+
+async fn init_server_icon_changer(ctx: impl serenity::CacheHttp, guild_id: serenity::GuildId) {
+	let mut interval = tokio::time::interval(Duration::from_secs(
+		rand::thread_rng().gen_range((24 * 60 * 60)..(48 * 60 * 60)),
```
Comment (conradludgate) [inline src/main.rs]: This won't choose a random time for each interval. I'd instead just use sleep inside the loop

File: src/main.rs [hunk 2]
```diff
@@ -209,5 +208,57 @@ async fn event_handler(
...
+	));
```
Comment (T0mstone) [inline src/main.rs]: Why not use `rand::distributions::Uniform<Duration>`?

Comment (ivandardi) [reply src/main.rs]: If I use that then the function becomes !Send.

File: src/main.rs [hunk 3]
```diff
@@ -209,5 +208,52 @@ async fn event_handler(
...
+async fn init_server_icon_changer(
+	ctx: impl serenity::CacheHttp,
+	guild_id: serenity::GuildId,
+) -> anyhow::Result<()> {
+	let icon_paths = std::fs::read_dir("./assets/server-icons")
+		.map_err(|e| anyhow!("Failed to read server-icons directory: {}", e))?
+		.filter_map(|entry| entry.ok())
+		.filter_map(|entry| {
+			let path = entry.path();
+			if path.is_file() {
```
Comment (bend-n) [inline src/main.rs:228]: what about `path.is_file().then_some(path)`

**Comment by pulustelu** [comment]
Needs to filter out the mourning icon, as that is inappropriate to select randomly

**Pull request merged**